common/authentication/postgresql/metadata.go

```diff
@@ -32,9 +35,11 @@ type PostgresAuthMetadata struct {
 	ConnectionMaxIdleTime time.Duration `mapstructure:"connectionMaxIdleTime"`
 	MaxConns              int           `mapstructure:"maxConns"`
 	UseAzureAD            bool          `mapstructure:"useAzureAD"`
+	UseAWSIAM             bool          `mapstructure:"useAWSIAM"`
```

Comment:
This line right here is what already automatically handles the IsTruthy check etc when the metadata gets unmarshalled into the struct using DecodeMetadata.

Only access metadata via this struct - do not manual access metadata except in testing!

Reply:
see https://github.com/dapr/components-contrib/pull/3324#discussion_r1563051803